ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -124,6 +255,20 @@ public String getLongForm() {
     StringBuilder sb = new StringBuilder();
     sb.append("[").append(getID());
     sb.append(":").append(version);
+    switch (version) {
+      case BLOB_ID_V1:
+        break;
+
+      case BLOB_ID_V2:
+        sb.append(":").append(flag);
+        sb.append(":").append(datacenterId);
+        sb.append(":").append(accountId);
+        sb.append(":").append(containerId);
+        break;
+
+      default:
+        throw new IllegalArgumentException("blob id version=" + version + " not supported");
+    }
```

Comment:
`getID()` is a bit expensive, not sure if this was the original reason not use it. Following the same way I could add `flag`, `datacenterId`, `accountId`, and `containerId`.